### ISSUE TITLE
Implement file history viewing, including local and Git.

### ISSUE THREAD

**Issue by liuhuo23** [zed-industries/zed#16827]
### Check for existing issues

- [X] Completed

### Describe the feature

Right-clicking on the edit page allows you to view the file's history of modifications and Git historical versions.

**Comment by JosephTLyons** [comment]
I think

- https://github.com/zed-industries/zed/issues/4268

... covers the need for local file history. Feel free to upvote that issue with a 👍.

**Comment by mirsella** [comment]
undotree like would be awesome https://github.com/jiaoshijie/undotree

**Comment by vitorfdl** [comment]
Really missing the local file history, coming from vscode.

**Comment by dominik1001** [comment]
This is the biggest issue keeping me from fully switching right now.

**Comment by DGolubets** [comment]
Yeah, every time I need to compare some changes I have to go back to VSCode. 
Git needs more love from Zed.

**Comment by ahmet-cetinkaya** [comment]
I'm waiting for this feature to switch to the Zed editor.

**Comment by onenewcode** [comment]
I really enjoy Zed, but I still prefer using a graphical interface like git-graph when working with Git.I hope Zed can roll out similar visual Git features sooner rather than later

**Comment by insane-dreamer** [comment]
I have to open PyCharm, when working with Zed, just for this feature.

**Comment by wzulfikar** [comment]
If you're fine with some basic (terminal) UI, you can have file history using combination of lazygit + zed tasks.json + keymap. Here's how I set it up:

1. Install [`lazygit`](https://github.com/jesseduffield/lazygit). We'll use `lazygit -f <filepath>` to show git history of a file.

https://github.com/user-attachments/assets/320bfcd6-815a-4f79-a73c-983ce423d53b

2. Add 'Show File History' in your tasks.json file. In my case (macos), it's in `~/config/zed/tasks.json`. I added it to my global tasks.json file because I know I'll use it everywhere (i.e. not project-specific).

<img width="600" height="192" alt="Image" src="https://github.com/user-attachments/assets/fb7a41f0-9d2a-4b35-82f6-dab89bf5810e" />


> Docs on tasks.json for ref: https://zed.dev/docs/tasks

3. Bind the task to a keymap. In my case, I added this entry to my `~/config/zed/keymap.json`:

<img width="600" height="301" alt="Image" src="https://github.com/user-attachments/assets/3ab28106-b0de-4663-8dd1-539f9af96d28" />

> If you want the task to run in the terminal dock instead of main editor, change the `"reveal_target"` from `"center"` to `"dock"`.

Now I can press `alt+shift+f` to show git history of current file. Demo:

https://github.com/user-attachments/assets/93e7f945-948b-43e4-b25c-ab2f83d05e5d

If this is okay for you, then it might be a good enough workaround while we wait for the native integration.

**Comment by insane-dreamer** [comment]
> If you're fine with some basic (terminal) UI, you can have file history using combination of lazygit + zed tasks.json + keymap. Here's how I set it up:

Nice setup! I'm a heavy lazygit user, and always have it open in another panel, but hadn't thought of launching it with a Zed shortcut. 

However, being able to do it within Zed would be so much better, allow you to click on a commit and see the diff in Zed, easily switch to other files, etc.

**Comment by cole-miller** [comment]
@ddoemonn is working on this!

**Comment by wzulfikar** [comment]
Wooo can’t wait to test it!

**Comment by esthertrapadoux** [comment]
@ddoemonn GitHub is weird but do you mind commenting on this issue so I can assign it to you? :)

**Comment by ddoemonn** [comment]
I would like to work on this!

**Comment by thernstig** [comment]
Tip: If doing this, implement <kbd>alt+,</kbd> and <kbd>alt+.</kbd> like-navigation to quickly jump between previous/next git commit file history. Pressing <kbd>alt+,</kbd> in a file immediately starts a diff with the previous version.

This inspiration comes from the GitLens extension in VS Code. There is a Github issue in VS Code to add this natively now.

**Comment by esthertrapadoux** [comment]
Thanks @ddoemonn ! Assigned this to you!

**Comment by HughHoyland** [comment]
While git graph panel (https://github.com/zed-industries/zed/pull/39595) is not exactly the same task, but still it might be beneficial to use the same component for the "git log" panel.

**Comment by ungario** [comment]
@ddoemonn is it somehow possible to view the local history as well? just like in vscode? or is vscode doing this with a separate database?
